### ISSUE TITLE
Possible to inform GitLab of pending job as soon as pipeline is triggered?

### ISSUE THREAD

**Issue by MarcelNehring** [jenkinsci/gitlab-plugin#783]
### Context
- **Gitlab plugin version**: 1.5.6
- **Gitlab version**: 10.8.4
- **Jenkins version**: 2.121.1
- **Job type**: Pipeline

### Problem description

This is a follow-up question to #420 

If I put `gitlabCommitStatus` after the checkout this also means that informing GitLab about the pending Jenkins job is postponed until after the checkout. If the pipeline has to wait for a node to become available or the checkout takes a long time users will not see in GitLab that a pipeline run is pending. Is there a workaround for this?

What confuses me a bit is the fact that a freestyle job does not seem to have this limitation. As soon as the Jenkins job is triggered by GitLab it reports the pending job back to GitLab. Also if I am not mistaken the commit hash of the revision is included in the JSON data GitLab posts to Jenkins via the webhook. Why do we have to wait for the checkout to know which revision is being build?

**Comment by semos** [comment]
You don't have to wait if your jenkinsfile is in same repo from the one that contains your code, as Jenkins already checked out and knows about the revision.

See https://github.com/jenkinsci/gitlab-plugin#scripted-pipeline-jobs
`NOTE: If you use Pipeline global libraries, or if you clone your project's Jenkinsfile from a repo different from the one that contains the relevant source code, you need to be careful about when you send project status. In short, make sure you put your gitlabCommitStatus or other similar steps after the SCM step that clones your project's source.`

For example, this is working flawlessly with jenkinsfile in code repo : 
```
gitlabCommitStatus("build") {
  stage('Checkout') {
    ...
  }
  stage('Images') {
    ...
  }
}
```

**Comment by MarcelNehring** [comment]
OK, thank you. Looks like I did not understand that paragraph correctly.
It takes several seconds for the `gitlabBuilds` step to become visible in GitLab.  I was just to impatient when I tried it. It indeed works. 👍 

--EDIT--
For future reference: My problem was that I put the `gitlabBuilds` command inside the `node` step. Thus GitLab was informed of the pending pipeline not before a node became available. If I move the `gitlabBuilds` step before any `node` step the Jenkins master executes the command immediately after the pipeline gets triggered and thus GitLab is informed immediately. 🎉
